packages/web3-providers-ws/src/index.ts

```diff
@@ -169,7 +169,8 @@ export default class WebSocketProvider<
 	}
 
 	public disconnect(code?: number, reason?: string): void {
-		this._removeSocketListeners();
+		// todo Is next line really needed? onClose will never be called
+		// this._removeSocketListeners();
 		this._webSocketConnection?.close(code, reason);
```

Comment:
I tried searching for the answer to my question but couldn't find it, will try it on my end later when i have more time but if someone has the answer:
On `close` does close remove event listeners or is that something we need to do manually?

Reply:
Maybe @nazarhussain could give us some insight on the topic.
Also, I would like to mention that `this._removeSocketListeners()` is also called on the `onclose` handler when we receive the event from the server.

Reply:
Okay, just uncomment `this._removeSocketListeners();` and it lgtm